envconfig.go

```diff
@@ -43,18 +43,30 @@ func (e *ParseError) Error() string {
 	return fmt.Sprintf("envconfig.Process: assigning %[1]s to %[2]s: converting '%[3]s' to type %[4]s", e.KeyName, e.FieldName, e.Value, e.TypeName)
 }
 
-// Process populates the specified struct based on environment variables
-func Process(prefix string, spec interface{}) error {
+// varInfo maintains information about the configuration variable
+type VarInfo struct {
```

Comment:
The idea is that this is public so that if people wanted to extend the capability to write their own loops that process the variables they could. For example, beyond usage I have written the capability to output the configuration values which can be valuable as part of a log. 

If this structure was not public this would not be possible, so I would rather keep them public.